pytorch_lightning/trainer/trainer.py

```diff
@@ -2287,6 +2289,47 @@ def terminate_on_nan(self, val: bool) -> None:
         )
         self._terminate_on_nan = val  # : 212
 
+    def _call_callbacks_on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> Dict[str, dict]:
```

Comment:
move this to where `_call_callback_hooks` is defined so they're next to each other. in a comment, could you describe why this is a dedicated function instead of being part of `_call_callback_hooks` ?